core/src/eval/macros.rs

```diff
@@ -65,21 +65,6 @@ macro_rules! push_u256 {
 	)
 }
 
-/// Pops top element of the stack and converts it to `usize`.
-///
-/// The top element **must** be not greater than `usize::MAX`.
-/// This non-local invariant is enforced by gas metering infrastructure.
-macro_rules! pop_usize {
```

Comment:
I would propose to leave this macro and just rename it to `pop_usize_checked` than we will be able to write only on line instead of two:
```rust
pop_u256!(state, len);
let len = as_usize_or_fail(len);
```
vs
```rust
pop_usize_checked!(len);
```

Reply:
You're proposing to deoptimize length check behavior. 
Before casting length to `usize` we check is `lenth.is_zero()`.
I do not accept that solution.